digitalocean/resource_digitalocean_project_resource.go

```diff
@@ -0,0 +1,145 @@
+package digitalocean
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceDigitalOceanProjectResource() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDigitalOceanProjectResourceCreate,
+		Read:   resourceDigitalOceanProjectResourceRead,
+		Delete: resourceDigitalOceanProjectResourceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"resource": {
+				Type:         schema.TypeString,
```

Comment:
I thought about that, but decided against it because of the complexity in constructing the ID for such a resource for the `resourceDigitalOceanProjectResourceRead` method. That method needs to reconstruct the `digialocean_project_resource` from just the ID which means storing the URNs in the ID.

For managing one resource, the ID (as written in the PR currently) is just PROJECT_ID:URN. If `digitalocean_project_resource` managed multiple URNs, then those URNs would either need to be stored in the key (creating a really long key if potentially a lot of URNs) or forcing the need to hash the set of URNs and store the hash in the key which would mean `resourceDigitalOceanProjectResourceRead` would not be able to reconstruct the `digitalocean_project_resource`.

Am I missing anything? This resource should coexist with `digitalocean_project`, manual project management, and potentially several other Terraform states managing project membership, so the solution seems to need to support piecemeal management of project membership.

Reply:
> the solution seems to need to support piecemeal management of project membership.

Very good point. Though I do wonder about the user experience of having to use many blocks to assign a handful of resources. 


The read method only needs to be able to entirely recreate the resource from the ID if it is being used to import the resource using an `ImportStatePassthrough`. So if we wanted to allow multiple resources to be assigned with one block while also allowing for non-exclusive management, we'd need to track the differences between the config and what is save in state. The read method would only save URNs that matched the config to state. Then the update method could do something like: https://play.golang.org/p/D75W84mnp-e 

The `vault_identity_group_policies` resource does something like this: https://github.com/terraform-providers/terraform-provider-vault/blob/master/vault/resource_identity_group_policies.go#L113

What do you think?

Reply:
From my first glance, that code looks pretty interesting and seems like the way to go. I'll look it over more thoroughly when I can and implement that technique here and see if it lets this resource manage project membership for multiple DigitalOcean resources.

Reply:
And implemented.